ls1intum/Artemis
# [Enhancement] Handle manual results on trigger-all

### Checklist

### Motivation and Context



Trigger-All currently does not handle manual results. 
Resolves #2213 

### Description


Whenever a new automatic result is generated and a manual or semi-automatic result exists, these two results are combined into a new semi-automatic result with:

- The automatic feedback from the automatic result
- The manual feedback from the manual result
- The assessor and state (draft / completed) from the manual result
- A new calculated score and scoreString

### Steps for Testing


1. Log in to Artemis
2. Create or use a programming exercise with manual review enabled.
3. Add one or two participations and manual assessments.
4. Change the grading config (test case weights, sca penalties) and press 'Trigger All'.
5. Wait for the build completion and check the results. 
   - They should consist of the unchanged manual feedback
   - combined with the updated automatic feedback
   - and have the correct score and score-string assigned.
 6. Check the results in all places they are shown: Course Details, Exercise Details, Code Editor, Scores Page (as Instructor)

### Test Coverage




 * ProgrammingExerciseGradingService: 95%

### Screenshots



No UI changes
